frontend/lib/services/teamService.dart

```diff
@@ -0,0 +1,192 @@
+    };
+
+    Response<String> response =
+        await dio.get(baseURL, queryParameters: queryParameters);
+
+    try {
+      final responseJson = json.decode(response.data!) as List;
+      List<Team> teams = responseJson.map((e) => Team.fromJson(e)).toList();
+      return teams;
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> createTeam(String name) async {
+    var body = {
+      "name": name,
+    };
+
+    Response<String> response = await dio.post(baseURL, data: body);
+
+    try {
+      return Team.fromJson(json.decode(response.data!));
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> getTeam(String id) async {
+    Response<String> response = await dio.get(baseURL + '/$id');
+
+    try {
+      return Team.fromJson(json.decode(response.data!));
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> deleteTeam(String id) async {
+    Response<String> response = await dio.delete(baseURL + '/$id');
+
+    try {
+      return Team.fromJson(json.decode(response.data!));
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> updateTeam(String id, String name) async {
+    var body = {"name": name};
+
+    Response<String> response = await dio.put(baseURL + '/$id', data: body);
+
+    try {
+      return Team.fromJson(json.decode(response.data!));
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> addTeamMember(String id, String member, String role) async {
+    var body = {
+      "member": member,
+      "role": role,
+    };
+
+    Response<String> response =
+        await dio.post(baseURL + '/$id' + '/members', data: body);
+
+    try {
+      return Team.fromJson(json.decode(response.data!));
+    } on SocketException {
+      throw DeckException('No Internet connection');
+    } on HttpException {
+      throw DeckException('Not found');
+    } on FormatException {
+      throw DeckException('Wrong format');
+    }
+  }
+
+  Future<Team?> updateTeamMemberRole(
+      String id, String memberID, Member member, String role) async {
```

Comment:
In this function, I'm not sure about the type of the member parameter. I checked the file ```teams-id-members-memberid.json``` but it did not specified the type of member, so I assumed it was an object.

Reply:
Swagger was fixed in the meantime and checking it reveals that `member` is the id of a member. Translating id's to actual objects is a matter for later so this is fine for now :+1: